ONSdigital/eq-cir-converter-service
# CIMS-4 - Install fastapi and spin up local Hello World application

### What is the context of this PR?

https://jira.ons.gov.uk/browse/CIMS-4
https://jira.ons.gov.uk/browse/CIMS-7

- Installed FastAPI, Uvicorn libraries
- Setup file structure for `main.py`
- Initial test framework with `pytest` and `test_main.py` with test coverage report
- Build docker image and spin up the docker container
- Spin up the local 'Hello World' application OR the docker container application
- Invoke the Swagger FastAPI endpoint to get the 'Hello World' response


### How to review

- Install all the packages using `poetry install` (please note that the poetry version must be 1.8.4)
- Start the local FastAPI application by running the command `make run`
      OR
- Start the docker container application by running the command `make docker-compose-up`
- Test the application by navigating to `http://127.0.0.1:5010/docs` or `http://localhost:5010/docs`


- The 'Hello World' application has been deployed in GCP Cloud Run - https://console.cloud.google.com/run?referrer=search&project=ons-cir-mig-dev
- The application is available in this URL - [cir-converter-service](https://cir-converter-service-418192618574.europe-west2.run.app/docs)
